03week/towersOfHanoi.js

```diff
@@ -18,28 +22,52 @@ function printStacks() {
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
-
-function movePiece() {
-  // Your code here
-
+/*The stacks object has 3 keys (a, b, c) and the key values 4, 3, 2, 1 are in an array.
+The .pop method will be used to remove the last element from the stacks array.
+Then the .push method will be used to place the removed value (block) that was popped
+and place it in any of the other stacks.*/
+const movePiece = (startStack, endStack) => {
+  stacks[endStack].push(stacks[startStack].pop());
 }
 
-function isLegal() {
+/*A move is legal is when a smaller block is placed in an empty stack by itself,
+ if a stack is occupied by a larger block then the move is not allowed. The .length
+ method is used to make the occupied stack equal to 0. */
+const isLegal = (startStack, endStack) => {
   // Your code here
-
+  if (stacks[endStack].length === 0) {
+    return true;
+  } else if (stacks[startStack].pop() < stacks[endStack].pop()) {
+    return true;
+  } else {
+    return false;
+  }
 }
-
-function checkForWin() {
+/*To check for win  a condition will need to be set up where all 4 of the blocks
+ will have to be moved from stack.a to either stack.b or stack.c. If this condition
+ is met then it will be returned as true. If neither condition is met then it is
+ a false conditional statement. The .length method will be used to determine if
+ all 4 blocks are stacked on stacks.b or stacks.c */
+const checkForWin = () => {
   // Your code here
-
+  if (stacks.c.length === 4) {
+    return true;
+  } else {
+    return false;
+  }
```

Comment:
This whole if/else could be simplified to `return stacks.c.length === 4 || stacks.b.length === 4`